### ISSUE TITLE
Broken asciidoctor.org asciidoctor-reveal.js page

### ISSUE THREAD

**Issue by obilodeau** [asciidoctor/asciidoctor.org#759]
Right now [the page](http://asciidoctor.org/docs/asciidoctor-revealjs/) is showing a stack trace, probably since we merged #755 :

```
undefined method `attr?' for nil:NilClass
Rendering file /home/travis/build/asciidoctor/asciidoctor.org/docs/asciidoctor-revealjs.adoc resulted in a failure.
Line: unknown

Backtrace:

/home/travis/build/asciidoctor/asciidoctor.org/_layouts/docs.html.slim:22:in `block in singleton class'
/home/travis/build/asciidoctor/asciidoctor.org/_layouts/docs.html.slim:1:in `instance_eval'
/home/travis/build/asciidoctor/asciidoctor.org/_layouts/docs.html.slim:1:in `singleton class'
/home/travis/build/asciidoctor/asciidoctor.org/_layouts/docs.html.slim:-1:in `__tilt_12242800'
/home/travis/build/asciidoctor/asciidoctor.org/.bundle/gems/ruby/2.4.0/gems/tilt-2.0.2/lib/tilt/template.rb:164:in `call'
/home/travis/build/asciidoctor/asciidoctor.org/.bundle/gems/ruby/2.4.0/gems/tilt-2.0.2/lib/tilt/template.rb:164:in `evaluate'
/home/travis/build/asciidoctor/asciidoctor.org/.bundle/gems/ruby/2.4.0/gems/tilt-2.0.2/lib/tilt/template.rb:99:in `render'
/home/travis/build/asciidoctor/asciidoctor.org/.bundle/gems/ruby/2.4.0/gems/awestruct-0.5.7/lib/awestruct/handlers/base_tilt_handler.rb:134:in `rendered_content'
/home/travis/build/asciidoctor/asciidoctor.org/.bundle/gems/ruby/2.4.0/gems/awestruct-0.5.7/lib/awestruct/handlers/layout_handler.rb:47:in `rendered_content'
/home/travis/build/asciidoctor/asciidoctor.org/.bundle/gems/ruby/2.4.0/gems/awestruct-0.5.7/lib/awestruct/page.rb:143:in `rendered_content'
/home/travis/build/asciidoctor/asciidoctor.org/.bundle/gems/ruby/2.4.0/gems/awestruct-0.5.7/lib/awestruct/handlers/layout_handler.rb:55:in `block in rendered_content'
/home/travis/build/asciidoctor/asciidoctor.org/.bundle/gems/ruby/2.4.0/gems/awestruct-0.5.7/lib/awestruct/handlers/layout_handler.rb:40:in `for_layout_chain'
/home/travis/build/asciidoctor/asciidoctor.org/.bundle/gems/ruby/2.4.0/gems/awestruct-0.5.7/lib/awestruct/handlers/layout_handler.rb:51:in `rendered_content'
/home/travis/build/asciidoctor/asciidoctor.org/.bundle/gems/ruby/2.4.0/gems/awestruct-0.5.7/lib/awestruct/page.rb:143:in `rendered_content'
/home/travis/build/asciidoctor/asciidoctor.org/.bundle/gems/ruby/2.4.0/gems/awestruct-0.5.7/lib/awestruct/engine.rb:427:in `generate_page'
/home/travis/build/asciidoctor/asciidoctor.org/.bundle/gems/ruby/2.4.0/gems/awestruct-0.5.7/lib/awestruct/engine.rb:403:in `block in generate_output'
/home/travis/build/asciidoctor/asciidoctor.org/.bundle/gems/ruby/2.4.0/gems/parallel-1.12.1/lib/parallel.rb:484:in `call_with_index'
/home/travis/build/asciidoctor/asciidoctor.org/.bundle/gems/ruby/2.4.0/gems/parallel-1.12.1/lib/parallel.rb:455:in `process_incoming_jobs'
/home/travis/build/asciidoctor/asciidoctor.org/.bundle/gems/ruby/2.4.0/gems/parallel-1.12.1/lib/parallel.rb:437:in `block in worker'
/home/travis/build/asciidoctor/asciidoctor.org/.bundle/gems/ruby/2.4.0/gems/parallel-1.12.1/lib/parallel.rb:428:in `fork'
/home/travis/build/asciidoctor/asciidoctor.org/.bundle/gems/ruby/2.4.0/gems/parallel-1.12.1/lib/parallel.rb:428:in `worker'
/home/travis/build/asciidoctor/asciidoctor.org/.bundle/gems/ruby/2.4.0/gems/parallel-1.12.1/lib/parallel.rb:419:in `block in create_workers'
/home/travis/build/asciidoctor/asciidoctor.org/.bundle/gems/ruby/2.4.0/gems/parallel-1.12.1/lib/parallel.rb:418:in `each'
/home/travis/build/asciidoctor/asciidoctor.org/.bundle/gems/ruby/2.4.0/gems/parallel-1.12.1/lib/parallel.rb:418:in `each_with_index'
/home/travis/build/asciidoctor/asciidoctor.org/.bundle/gems/ruby/2.4.0/gems/parallel-1.12.1/lib/parallel.rb:418:in `create_workers'
/home/travis/build/asciidoctor/asciidoctor.org/.bundle/gems/ruby/2.4.0/gems/parallel-1.12.1/lib/parallel.rb:358:in `work_in_processes'
/home/travis/build/asciidoctor/asciidoctor.org/.bundle/gems/ruby/2.4.0/gems/parallel-1.12.1/lib/parallel.rb:264:in `map'
/home/travis/build/asciidoctor/asciidoctor.org/.bundle/gems/ruby/2.4.0/gems/awestruct-0.5.7/lib/awestruct/engine.rb:401:in `generate_output'
/home/travis/build/asciidoctor/asciidoctor.org/.bundle/gems/ruby/2.4.0/gems/awestruct-0.5.7/lib/awestruct/engine.rb:121:in `run'
/home/travis/build/asciidoctor/asciidoctor.org/.bundle/gems/ruby/2.4.0/gems/awestruct-0.5.7/lib/awestruct/cli/generate.rb:22:in `run'
/home/travis/build/asciidoctor/asciidoctor.org/.bundle/gems/ruby/2.4.0/gems/awestruct-0.5.7/lib/awestruct/cli/invoker.rb:150:in `invoke_generate'
/home/travis/build/asciidoctor/asciidoctor.org/.bundle/gems/ruby/2.4.0/gems/awestruct-0.5.7/lib/awestruct/cli/invoker.rb:74:in `invoke!'
/home/travis/build/asciidoctor/asciidoctor.org/.bundle/gems/ruby/2.4.0/gems/awestruct-0.5.7/bin/awestruct:9:in `'
/home/travis/build/asciidoctor/asciidoctor.org/.bundle/gems/ruby/2.4.0/bin/awestruct:23:in `load'
/home/travis/build/asciidoctor/asciidoctor.org/.bundle/gems/ruby/2.4.0/bin/awestruct:23:in `'
/home/travis/.rvm/rubies/ruby-2.4.1/lib/ruby/site_ruby/2.4.0/bundler/cli/exec.rb:75:in `load'
/home/travis/.rvm/rubies/ruby-2.4.1/lib/ruby/site_ruby/2.4.0/bundler/cli/exec.rb:75:in `kernel_load'
/home/travis/.rvm/rubies/ruby-2.4.1/lib/ruby/site_ruby/2.4.0/bundler/cli/exec.rb:28:in `run'
/home/travis/.rvm/rubies/ruby-2.4.1/lib/ruby/site_ruby/2.4.0/bundler/cli.rb:424:in `exec'
/home/travis/.rvm/rubies/ruby-2.4.1/lib/ruby/site_ruby/2.4.0/bundler/vendor/thor/lib/thor/command.rb:27:in `run'
/home/travis/.rvm/rubies/ruby-2.4.1/lib/ruby/site_ruby/2.4.0/bundler/vendor/thor/lib/thor/invocation.rb:126:in `invoke_command'
/home/travis/.rvm/rubies/ruby-2.4.1/lib/ruby/site_ruby/2.4.0/bundler/vendor/thor/lib/thor.rb:387:in `dispatch'
/home/travis/.rvm/rubies/ruby-2.4.1/lib/ruby/site_ruby/2.4.0/bundler/cli.rb:27:in `dispatch'
/home/travis/.rvm/rubies/ruby-2.4.1/lib/ruby/site_ruby/2.4.0/bundler/vendor/thor/lib/thor/base.rb:466:in `start'
/home/travis/.rvm/rubies/ruby-2.4.1/lib/ruby/site_ruby/2.4.0/bundler/cli.rb:18:in `start'
/home/travis/.rvm/rubies/ruby-2.4.1/lib/ruby/gems/2.4.0/gems/bundler-1.16.0/exe/bundle:30:in `block in '
/home/travis/.rvm/rubies/ruby-2.4.1/lib/ruby/site_ruby/2.4.0/bundler/friendly_errors.rb:122:in `with_friendly_errors'
/home/travis/.rvm/rubies/ruby-2.4.1/lib/ruby/gems/2.4.0/gems/bundler-1.16.0/exe/bundle:22:in `'
/home/travis/.rvm/gems/ruby-2.4.1/bin/bundle:23:in `load'
/home/travis/.rvm/gems/ruby-2.4.1/bin/bundle:23:in `
'
/home/travis/.rvm/gems/ruby-2.4.1/bin/ruby_executable_hooks:15:in `eval'
/home/travis/.rvm/gems/ruby-2.4.1/bin/ruby_executable_hooks:15:in `
'
```

Travis-CI's successfull (?) build gives more details: https://travis-ci.org/asciidoctor/asciidoctor.org/builds/321443627#L630

I'm going to do an emergency release in the following hours or days due to asciidoctor/asciidoctor-reveal.js#172 so if I need to change something on my side let me know.

First reported by @Maurice-Mueller here: asciidoctor/asciidoctor-reveal.js#172

**Comment by mojavelinux** [comment]
It looks like a broken include. I'll check it out.

(Interesting to note, Travis builds were failing on New Years Eve, so we were getting some false negatives).

**Comment by mojavelinux** [comment]
I can confirm that the problem was that the include path was incorrect. But the extension code I wrote was too fragile and was crashing the processor instead of issuing a warning. I've updated the extension code. Now it shows that the include path is incorrect.

**Comment by obilodeau** [comment]
Content is now visible but at some point the broken include kicks in. From there content is confused probably mixing example asciidoc markup that should have been in a source listing but that is not anymore.

I'll merge asciidoctor/asciidoctor-reveal.js#173 soon and send a PR to update README to 1.1.1 once released. We'll see how it looks like from there.

**Comment by mojavelinux** [comment]
It looks like there is another error in my extension code. I'll get it
fixed later today. Stay tuned!

**Comment by mojavelinux** [comment]
Now it's fixed for real. I finally figured out how to use my own API ;)

I'm pretty sure if you fixed the includes in the README everything will be working perfectly once we update the reference in asciidoctor.org.

**Comment by mojavelinux** [comment]
resolved by e261ed677b23416ae4f538e9177eed4c8a1524ec

**Comment by mojavelinux** [comment]
Btw, the functionality this extension provides will be implemented natively in Asciidoctor 1.5.7.
